kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateList.js

```diff
@@ -75,62 +94,19 @@ define([
 
         function start(arg) {
             return Promise.try(function() {
-                container = arg.node;
-                container.classList.add(`${cssBaseClass}__container`);
-                container.classList.add('batch-mode-list');
+                container = $(arg.node);
+                container.addClass([`${cssBaseClass}__container`, 'batch-mode-list']);
                 UI.make({ node: container });
-                container.innerHTML = renderTable();
-                parentJobId = arg.parentJobId;
+                container.append($(createTable()));
 
                 return Promise.try(() => {
-                    createJobStateWidget('parent', parentJobId, model.getItem('exec.jobState.status'), arg.clickFunction, true);
-                    container.getElementsByTagName('tr')[0].classList.add(selectedJobCssClass); // start with the parent selected
-
-                    for (let i=0; i<Math.max(arg.batchSize, arg.childJobs.length); i++) {
-                        let jobId = null,
-                            initialState = null;
-                        if (i < arg.childJobs.length) {
-                            jobId = arg.childJobs[i].job_id;
-                            initialState = arg.childJobs[i].status;
-                        }
-                        createJobStateWidget(i, jobId, initialState, arg.clickFunction);  // can make null ones. these need to be updated.
-                    }
-                })
-                    .then(() => { startParentListener(); });
-            });
-        }
-
-        function startParentListener() {
-            parentListener = runtime.bus().listen({
-                channel: {
-                    jobId: parentJobId
-                },
-                key: {
-                    type: 'job-status'
-                },
-                handle: handleJobStatusUpdate
-            });
-        }
-
-        /**
-         * Pass the job state to all row widgets, if it exists.
-         * @param {Object} message
-         */
-        function handleJobStatusUpdate(message) {
-            if (message.jobState.batch_size && message.jobState.child_jobs.length === 0) {
-                if (message.jobState.job_output.result[0].batch_results) {
-                    message.jobState.child_jobs = Object.keys(message.jobState.job_output.result[0].batch_results)
-                        .map((item) => {
-                            return message.jobState.job_output.result[0].batch_results[item].final_job_state;
-                        });
-                }
-            }
-
-            if (message.jobState.child_jobs) {
-                message.jobState.child_jobs.forEach((state, idx) => {
-                    widgets[idx].updateState(state);
+                    arg.childJobs.forEach((childJob, index) => {
+                        createJobStateWidget(index, childJob.job_id, childJob.status);
```

Comment:
do these need to complete before we render the container? if so i might recommend returning a promise from createJobStateWidget and changing this to be a Promise.all call (instead of Promise.try)